packages/app-check/src/internal-api.ts

```diff
@@ -264,7 +277,11 @@ function createTokenRefresher(
 
       // getToken() always resolves. In case the result has an error field defined, it means the operation failed, and we should retry.
       if (result.error) {
-        throw result.error;
+        if (onError) {
+          onError(result.error);
```

Comment:
This is redundant. Notifying listeners are already handled in `getToken()` which calls `notifyTokenListeners()`.

Here we should always throw to trigger retries. (Just revert the changes)

Reply:
Reverted.